airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/source.py

```diff
@@ -0,0 +1,80 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from datetime import datetime
+from typing import Generator
+from msal.exceptions import MsalServiceError
+
+from .client import Client
+
+from airbyte_protocol import AirbyteCatalog, AirbyteConnectionStatus, \
+    AirbyteMessage, Status, ConfiguredAirbyteCatalog, Type, AirbyteRecordMessage
+from base_python import AirbyteLogger, ConfigContainer, Source
+
+
+class SourceMicrosoftTeams(Source):
```

Comment:
A class was created that reduces the amount of code to create a new source: `BaseSource`. Consider inheriting from it.